flink-table/flink-table-planner-loader/src/main/java/org/apache/flink/table/planner/loader/PlannerModule.java

```diff
@@ -150,4 +160,30 @@ public PlannerFactory loadPlannerFactory() {
         return FactoryUtil.discoverFactory(
                 this.submoduleClassLoader, PlannerFactory.class, PlannerFactory.DEFAULT_IDENTIFIER);
     }
+
+    /**
+     * A class loader extending {@link ComponentClassLoader} which overwrites method{@link #addURL}
+     * to enable it can add url to component classloader.
+     */
+    private static class PlannerComponentClassLoader extends ComponentClassLoader {
```

Comment:
Why not just use `ComponentClassLoader`? It doesn't seem to extend anything.

Reply:
We overwride the method `addURL` to make it public